master/src/SpotlightPanel.cpp

```diff
@@ -108,6 +108,12 @@ void SpotlightPanel::add()
 		// due to a bug in QListView force relayout of all items to show decorations (thumbnails) properly
 		updateIconSize();
 	}
+
+	this->show();
+	auto m_master = VeyonCore::instance()->findChild<VeyonMasterInterface *>();
+	auto window = m_master->mainWindow();
+	auto panelButton = window->findChild<QToolButton *>( QStringLiteral("spotlightPanelButton") );
```

Comment:
I suggest to remove the first 2 lines and simplify via

`auto panelButton = window()->findChild<QToolButton *>( QStringLiteral("spotlightPanelButton") ); `

since the spotlight panel widget is a child of the main window and so can access the window via `QWidget::window()` directly ;-)

We could even improve this further instead by adding a `visibilityChanged()` signal to `SpotlightPanel` and connect it to the `setChecked()` method of the corresponding button directly in MainWindow so we don't add implicit assumptions about the internal name of a button (which might change in the future and will not cause a compiler error). Is that enough information?

Reply:
I hope so an will try to improve it as suggested. Thank you. :+1: